src/ImageSharp/Common/Helpers/SimdUtils.HwIntrinsics.cs

```diff
@@ -342,11 +342,11 @@ public static class HwIntrinsics
                 ref Vector128<byte> destBase =
                     ref Unsafe.As<byte, Vector128<byte>>(ref MemoryMarshal.GetReference(dest));
 
-                int n = dest.Length / Vector128<byte>.Count;
-                int m = Numerics.Modulo4(n);
-                int u = n - m;
+                nint n = (nint)(uint)dest.Length / Vector128<byte>.Count;
+                nint m = Numerics.Modulo4(n);
+                nint u = n - m;
 
-                for (int i = 0; i < u; i += 4)
+                for (nint i = 0; i < u; i += 4)
```

Comment:
Because of the `n - m` the loop with `nint` is kept. With `nuint` one has to take care of underflow due the subtraction.

Alternatives with using `nuint` would be casting to a signed type or having a `if + do-while` looping construct. I think the `nint` as used here is simpler.